creator-node/src/services/stateMachineManager/stateMonitoring/index.js

```diff
@@ -38,75 +37,94 @@ const cNodeEndpointToSpIdMapQueueLogger = createChildLogger(baseLogger, {
  */
 class StateMonitoringManager {
   async init(discoveryNodeEndpoint, prometheusRegistry) {
-    // Create and start queue to fetch cNodeEndpoint->spId mapping
-    const cNodeEndpointToSpIdMapQueue = makeQueue({
+    // Create queue to fetch cNodeEndpoint->spId mapping
+    const { queue: cNodeEndpointToSpIdMapQueue } = makeQueue({
       name: QUEUE_NAMES.FETCH_C_NODE_ENDPOINT_TO_SP_ID_MAP,
+      processor: this.makeProcessJob(
+        fetchCNodeEndpointToSpIdMapJobProcessor,
+        cNodeEndpointToSpIdMapQueueLogger,
+        prometheusRegistry
+      ).bind(this),
+      logger: cNodeEndpointToSpIdMapQueueLogger,
       removeOnComplete: QUEUE_HISTORY.FETCH_C_NODE_ENDPOINT_TO_SP_ID_MAP,
       removeOnFail: QUEUE_HISTORY.FETCH_C_NODE_ENDPOINT_TO_SP_ID_MAP,
-      lockDuration: MAX_QUEUE_RUNTIMES.FETCH_C_NODE_ENDPOINT_TO_SP_ID_MAP,
       prometheusRegistry,
       limiter: {
         max: 1,
         duration: config.get('fetchCNodeEndpointToSpIdMapIntervalMs')
+      },
+      onFailCallback: (job, error, prev) => {
+        cNodeEndpointToSpIdMapQueueLogger.error(
+          `Queue Job Failed - ID ${job?.id} - Error ${error}`
+        )
+        cNodeEndpointToSpIdMapQueue.add('first-job', {})
```

Comment:
add `first-job` on fail?

Reply:
i dont think i followall the changes to this file - maybe can discuss on call

Reply:
discussed on call and changed that name from `first-job` to `retry-on-fail` - that was a typo